packages/shared-internal/src/dtos/step-schemas/generate-preview-response.dto.ts

```diff
@@ -0,0 +1,103 @@
+import { z } from 'zod';
+import { ChannelTypeEnum } from '@novu/shared';
+import { ControlPreviewIssueType } from './control-preview-issue.type';
+
+export enum RedirectTargetEnum {
+  SELF = '_self',
+  BLANK = '_blank',
+  PARENT = '_parent',
+  TOP = '_top',
+  UNFENCED_TOP = '_unfencedTop',
+}
+
+// Define the Zod schemas for each preview result class
+export const PreviewResultSchema = z.object({});
```

Comment:
Don't we have result schemas already in @novu/framework? If not, they definitely belong there.

Reply:
it needs to be in shared, I cannot try to put something in shared dependent on the framwork

Reply:
What do you think about putting these in application-generic? If not mistaken, this code is used between the API and the worker. That way we avoid bloating @novu/shared and we are still sharing code using application-generic.

Reply:
it's supposed to be available in the client

Reply:
it's a DTO returning from the server, they need it